tests/components/justnimbus/test_config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Test the JustNimbus config flow."""
+from unittest.mock import patch
+
+from justnimbus.exceptions import InvalidClientID, JustNimbusError
+
+from homeassistant import config_entries
+from homeassistant.components.justnimbus.const import DOMAIN
+from homeassistant.const import CONF_CLIENT_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch("justnimbus.JustNimbusClient.get_data", return_value=True,), patch(
+        "homeassistant.components.justnimbus.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_CLIENT_ID: "test_id",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "JustNimbus"
+    assert result2["data"] == {
+        CONF_CLIENT_ID: "test_id",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_client_id = "test_id"
+
+    with patch(
+        "justnimbus.JustNimbusClient.get_data",
+        side_effect=InvalidClientID(client_id="test_id"),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_CLIENT_ID: mock_client_id,
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "justnimbus.JustNimbusClient.get_data",
+        side_effect=JustNimbusError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_CLIENT_ID: "test_id",
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
```

Comment:
This is the exact same tests as above, except with a different side effect and resulting error. Instead of writing the same test twice, you could use parameterize it using pytest.